mimium-lang/src/compiler/mirgen.rs

```diff
@@ -422,32 +405,16 @@ impl Context {
                         inst_refs.push(self.get_current_basicblock().0.len());
                         self.push_inst(Instruction::GetElement {
                             value: dst.clone(),
-                            ty: Type::Unknown.into_id(), // lazyly set after loops
+                            ty, // lazyly set after loops
```

Comment:
This `ty` is overwritten by `let (v, ty) = self.eval_expr(*e)?;`. It seems all the tests pass (except for `state_tuple`) by changing the variable name to avoid conflict.

``` patch
index 1c1dac6..160d6af 100644
--- a/mimium-lang/src/compiler/mirgen.rs
+++ b/mimium-lang/src/compiler/mirgen.rs
@@ -398,7 +398,7 @@ impl Context {
                 let dst = self.gen_new_register();
                 let mut inst_refs: Vec<usize> = vec![];
                 for (i, e) in items.iter().enumerate() {
-                    let (v, ty) = self.eval_expr(*e)?;
+                    let (v, elem_ty) = self.eval_expr(*e)?;
                     let ptr = if i == 0 {
                         dst.clone()
                     } else {
@@ -410,7 +410,7 @@ impl Context {
                             tuple_offset: i as u64,
                         })
                     };
-                    self.push_inst(Instruction::Store(ptr, v, ty));
+                    self.push_inst(Instruction::Store(ptr, v, elem_ty));
                 }
                 self.get_current_basicblock()
                     .0
```